microservice/challenge-1/app42/backend/src/repository/UserRepository.ts

```diff
@@ -0,0 +1,19 @@
+import User from "../entity/User";
+import IRepository from "./IRepository";
+
+export default class UserRepository implements IRepository<User> {
+
+    public add(user: User): Promise<User> {
```

Comment:
How and where would be the best place to validate an email upon user creation?

Reply:
There is isValid method in UserUtil.ts which is responsible of validation but it is not implemented.
It has been used in UserService.

Reply:
It would be better to check user validation in Controller and send the proper message if it's not valid.